routes/api/v1/favorites.js

```diff
@@ -0,0 +1,75 @@
+var express = require("express");
+var router = express.Router();
+const fetch = require('node-fetch');
+var pry = require("pryjs");
+require("dotenv").config();
+var Location = require('../../../models').Location;
+var User = require('../../../models').User;
+var UserLocation = require('../../../models').UserLocation;
+
+router.post("/", function(req, res, next) {
+  if (req.body.api_key) {
+    User.findAll(
+      {
+        where: {
+          api_key: req.body.api_key
+        }
+      }
+    )
+      .then(user => {
+        if (user.length === 0) {
+          res.setHeader("Content-Type", "application/json");
+          res.status(401).send("Unauthorized access")
+        } else {
+          Location.findAll(
+            {
+              where: {
+                location: req.body.location
+              }
+            }
+          )
+            .then(location => {
+              var favorite;
+              if (location.length === 0) {
```

Comment:
I like that you're only calling the Geocoding api if the location does not exist.